sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/Clauses/SelectClause.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.DigitalTwins.Core.QueryBuilder
+{
+    /// <summary>
+    /// Custom object for a SELECT clause. Only meant to be used when adding SELECT to a query. Hidden from user.
+    /// </summary>
+    internal class SelectClause : BaseClause
+    {
+        /// <summary>
+        /// The argument for the SELECT clause (eg. *).
+        /// </summary>
+        public string ClauseArg { get; set; }
+
+        /// <summary>
+        /// Constructor for SELECT clause.
+        /// </summary>
+        /// <param name="argument"> Argument for what to select (collection, property, etc.). </param>
+        public SelectClause(string argument)
```

Comment:
nit: even though this is more of an implementation comment, but I noticed that your select clause only takes one string as the argument, it appears to me that you may want to support an array of arguments ... something to think about.